processor/src/test/java/org/mapstruct/ap/test/bugs/_1801/domain/ImmutableItem.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._1801.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Immutable implementation of {@link Item}.
+ * <p>
+ * Superclass should expose a static subclass of the Builder to create immutable instance
+ * {@code public static Builder extends ImmutableItem.Builder}.
+ */
+@SuppressWarnings({"all"})
```

Comment:
@SuppressWarnings({"all}) ? It is really necessary? Maybe should we exclude some part?

Reply:
This class is identical to the one generated by `Immutables`. That's why the supress is there I think.

Reply:
did some polishing.. It still captures the essence but fixes suppress warnings